### PR TITLE
PR MobileNetV1

### PR DESCRIPTION
**Pull request by zkdlfrlwl2**
First PR

### CODE REVIEW

File: MobileNetV1/config.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+project_name: 'MobileNetV1'
+save_dir: './experiments'
+dataset: 'CIFAR10'
+data_dir: './data'
+torchscript_model_save_path: './experiments/'
+seed: '2022'
```
Comment (rapidrabbit76) [inline MobileNetV1/config.yaml:6]: seed str value 사용 가능하나??

File: MobileNetV1/model/block.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        x = self.bn(x)
+
+        return self.act(x)
+
+
+class DepthWiseSeparableBlock(nn.Module):
+
+    def __init__(
+            self,
+            dim: list,
```
Comment (rapidrabbit76) [inline MobileNetV1/model/block.py]: list -> List[int]

File: MobileNetV1/model/block.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                groups=dim[1]
+            )),
+            layers.append(ConvBlock(
+                in_channels=dim[1],
+                out_channels=dim[2],
+                kernel_size=1,
+                stride=stride[3]
+            ))
+
+        self.blocks = nn.Sequential(*layers)
```
Comment (rapidrabbit76) [inline MobileNetV1/model/block.py:130]: list 넣는 방식 좋아하네

Comment (rapidrabbit76) [reply MobileNetV1/model/block.py:130]: 난 그냥 identity 로 퉁처버리는 경우가 많은데

File: MobileNetV1/train.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    )
+
+    # Logger
+    wandb_logger = WandbLogger(
+        name=f'{config.project_name}-{config.dataset}',
+        project=config.project_name,
+        save_dir=config.save_dir,
+        log_model='all',
+    )
+    wandb_logger.experiment.config.update(hparams)
```
Comment (rapidrabbit76) [inline MobileNetV1/train.py:129]: self.save_hyperparameters() 하면 다 보관 되는거 아님 ??

**Review by rapidrabbit76 — CHANGES_REQUESTED** [review]

File: MobileNetV1/model/block.py (hunk 2 above)
Comment (zkdlfrlwl2) [reply MobileNetV1/model/block.py]: 접수

File: MobileNetV1/config.yaml (hunk 1 above)
Comment (zkdlfrlwl2) [reply MobileNetV1/config.yaml:6]: lightning 코드 까보니까 seed가 int type이 아니면 seed = int(seed) 이렇게 형 변환을 자체적으로 해주네 

`
elif not isinstance(seed, int):
    seed = int(seed)
`

File: MobileNetV1/train.py (hunk 4 above)
Comment (zkdlfrlwl2) [reply MobileNetV1/train.py:129]: wandb_logger.experiment.config.update(hparams) < 이 부분 말하는거가 ? 내가 디버깅 했을 때 self 확인해보니까 생성자에 self.save_hyperparameters() 넣어서 저장되는 건 생성자에서 입력으로 받는 파라미터만 저장되는 것 같던데 예를 들면 여기서 저장되는 건 image_channels, n_classes, lr, alpha 라는 소리

File: MobileNetV1/model/block.py (hunk 3 above)
Comment (zkdlfrlwl2) [reply MobileNetV1/model/block.py:130]: 저기서 어떻게 identity로 퉁침 ?

Comment (zkdlfrlwl2) [reply MobileNetV1/model/block.py:130]: list로 layer 쌓으니까 깔끔한 느낌이 들어서

File: MobileNetV1/train.py (hunk 4 above)
Comment (zkdlfrlwl2) [reply MobileNetV1/train.py:129]: 그리고 저 self.save_hyperparameters()로 저장되는게 자동적으로 wandb에 logging 되는지는 모르겠다 ? logging 되려면 저렇게 wandb_logger.experiment.config.update(hparams)를 따로 명시해줘야 하는거 아닌감 ?

File: MobileNetV1/config.yaml (hunk 1 above)
Comment (rapidrabbit76) [reply MobileNetV1/config.yaml:6]: 맞네 당연히 인트 넣어야라고 생각했었음

File: MobileNetV1/train.py (hunk 4 above)
Comment (rapidrabbit76) [reply MobileNetV1/train.py:129]: 아 맞제 config yaml 구조 바꿔야 할거같다. 
내부에 hparams 하나 더 놓는게 명확할거 같다고 생각했는데 돌리다 보니깐 아닌거 같더라고 ...

Comment (zkdlfrlwl2) [reply MobileNetV1/train.py:129]: 모델 생성자에 던져줘야하는 parameter하고 wandb에 따로 logging할 hyper parameter 이렇게 나눠서 config.yaml에 작성하면 될 것 같은데

File: MobileNetV1/config.yaml (hunk 1 above)
Comment (zkdlfrlwl2) [reply MobileNetV1/config.yaml:6]: 이런거 보면 라이브러리가 참 친절해 ㅋㅋ

File: MobileNetV1/train.py (hunk 4 above)
Comment (rapidrabbit76) [reply MobileNetV1/train.py:129]: > 모델 생성자에 던져줘야하는 parameter하고 wandb에 따로 logging할 hyper parameter 이렇게 나눠서 config.yaml에 작성하면 될 것 같은데

싫어 이러면 두개 관리해야해서 싫다 ...

Comment (zkdlfrlwl2) [reply MobileNetV1/train.py:129]: wandb table 보면 싹다 저장되는 것 같긴한데 어느 파트가 저장되는 건지는 좀 애매하네

**Comment by zkdlfrlwl2** [comment]
이런 PR 진행 중에 merge 하면 이것까지 포함되버리네 ㅋㅋ

File: MobileNetV1/config.yaml (hunk 1 above)
Comment (rapidrabbit76) [reply MobileNetV1/config.yaml:6]: issue #5  확인

**Review by rapidrabbit76 — COMMENTED** [review]
merge MobileNet
